general/mem_manager.cpp

```diff
@@ -352,8 +354,16 @@ class MmuHostMemorySpace : public HostMemorySpace
    MmuHostMemorySpace(): HostMemorySpace() { MmuInit(); }
    void Alloc(void **ptr, size_t bytes) { MmuAlloc(ptr, bytes); }
    void Dealloc(void *ptr) { MmuDealloc(ptr, maps->memories.at(ptr).bytes); }
-   void Protect(const void *ptr, size_t bytes) { MmuProtect(ptr, bytes); }
-   void Unprotect(const void *ptr, size_t bytes) { MmuAllow(ptr, bytes); }
+   void Protect(const Memory& mem, size_t bytes)
+   {
+      if (mem.h_rw) { MmuProtect(mem.h_ptr, bytes); }
+      mem.h_rw = false;
```

Comment:
Small optimization: move `mem.h_rw = false;` inside the `if` statement to avoid the memory write -- I don't expect the compiler to do this automatically even with optimization but I might be wrong.

This can also be done in the other 3 places where `MmuAllow` or `MmuProtect` are called.

Reply:
`done`